parsl/dataflow/future_resolution.py

```diff
@@ -26,23 +27,28 @@ def _(fut: Future):
 
 
 # The above is the traditional Parsl future/non-future behaviour.
-# Below is an example of shallow traversal of a tuple.
+# Below is an example of shallow traversal of iterables.
+
 
 @traverse_to_gather.register
-def _(tup: tuple):
+def _(iterable: Union[tuple, list, set]):
     # a "deep" traversal would instead recursively call traverse_to_gather
-    # here to inspect whatever is inside the tuple
-    return [v for v in tup if isinstance(v, Future)]
+    # here to inspect whatever is inside the sequence
+
+    return [v for v in iterable if isinstance(v, Future)]
 
 
 @traverse_to_unwrap.register
 @singledispatch
-def _(tup: tuple):
+def _(iterable: Union[tuple, list, set]):
```

Comment:
at least with set, there is some subtlety that I think means this should be separately implemented for set:

an iterable of mapping .result() over a set behaves differently to creating a set of the unwrapped result and returning that: the set {f(1), f(2)} only has one element in some situations and this code should use `set()` rather than discarding the set semantics.

Reply:
actually I see what you're trying to do with invoking runtime-discovered constructors...